src/data-type.js

```diff
@@ -137,11 +154,18 @@ const TYPE = module.exports.TYPE = {
     },
 
     writeParameterData: function(buffer, parameter) {
-      if (parameter.value != null) {
-        buffer.writeUInt8(4);
-        buffer.writeInt32LE(parseInt(parameter.value));
-      } else {
+      const value = parameter.value;
+
+      if (value === undefined || value === null) {
         buffer.writeUInt8(0);
+        return;
+      }
+
+      buffer.writeUInt8(4);
+      if (typeof value === 'number') {
```

Comment:
If it's a request parameter and the value is the number 0.000000006, the `parseInt()` call in the corresponding `validate()` function below will still convert it to 6, because in `Request.validateParameters()` the return value of validate() is used to replace the original parameter value.